lib/gemstash/gem_source/upstream_source.rb

```diff
@@ -160,7 +160,15 @@ def fetch_local_gem(gem_name, gem_resource, resource_type)
       def fetch_remote_gem(gem_name, gem_resource, resource_type)
         log.info "Gem #{gem_name.name} is not cached, fetching #{resource_type}"
         gem_fetcher.fetch(gem_name.id, resource_type) do |content, properties|
-          gem_resource.save({ resource_type => content }, headers: { resource_type => properties })
+          resource_properties = {
+            upstream: upstream.to_s,
+            gem_name: gem_name.name,
+            headers: { resource_type => properties }
+          }
+
+          gem = gem_resource.save({ resource_type => content }, resource_properties)
+          Gemstash::DB::CachedRubygem.store(upstream, gem_name, resource_type)
```

Comment:
I'm ok with it. It's just that class (or static) methods usually resist refactoring basically because they are global.

Not a blocker, just a thought.